### PR TITLE
[FIX] mail: messages from unknown thread not displaying on sidebar

### PR DESCRIPTION
**Pull request by tsm-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/composer/suggestion_service.js [hunk 1]
```diff
@@ -219,8 +219,8 @@ class SuggestionService {
                 return 1;
             }
             if (thread?.serverData?.channel) {
```
Comment (seb-odoo) [inline addons/mail/static/src/new/composer/suggestion_service.js]: Condition should be adapted too?

File: addons/mail/static/src/new/composer/suggestion_service.js [hunk 2]
```diff
@@ -219,8 +219,8 @@ class SuggestionService {
...
-                const isMember1 = thread.serverData.channel.channelMembers[0][1].includes(p1);
-                const isMember2 = thread.serverData.channel.channelMembers[0][1].includes(p2);
+                const isMember1 = thread.channel.channelMembers[0][1].includes(p1);
```
Comment (seb-odoo) [inline addons/mail/static/src/new/composer/suggestion_service.js:222]: I don't see where `channel` is defined on `thread` model actually.

Comment (tsm-odoo) [reply addons/mail/static/src/new/composer/suggestion_service.js:222]: `threadService#479`

Comment (seb-odoo) [reply addons/mail/static/src/new/composer/suggestion_service.js:222]: It should be added to thread model field definition too then.

And I would personally be in favor of defining a channel model, but I think that's more controverted.

File: addons/mail/static/src/new/core/thread_service.js [hunk 3]
```diff
@@ -475,7 +474,10 @@ export class ThreadService {
                 "message_needaction_counter",
                 "name",
                 "state",
+                "group_based_subscription",
             ]);
+            thread.channel = thread.channel ?? {};
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_service.js]: I feel there should be conditions on it, and it shouldn't be defined at all for non channel threads.

File: addons/mail/static/src/new/composer/suggestion_service.js (hunk 2 above)
Comment (tsm-odoo) [reply addons/mail/static/src/new/composer/suggestion_service.js:222]: Added in model field definitions, channel model as a follow up if we decide its the thing to do ?

File: addons/mail/static/src/new/core/thread_model.js [hunk 4]
```diff
@@ -30,6 +30,8 @@ export class Thread {
     attachments = [];
     /** @type {integer} */
     activeRtcSessionId;
+    /** @type {object|undefined} */
+    channel;
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js:34]: The problem with this syntax (and no model) is that we don't have autocomplete or any idea of what keys are inside channel. If we don't make the model, I would at least expect to have a typedef.

File: addons/mail/static/src/new/core/thread_model.js [hunk 5]
```diff
@@ -201,7 +203,7 @@ export class Thread {
     }
 
     get imgUrl() {
-        const avatarCacheKey = this.serverData?.channel?.avatarCacheKey;
+        const avatarCacheKey = this.channel.avatarCacheKey;
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js:206]: possibly need check on existence of channel

**Review by seb-odoo — APPROVED** [review]

**Pull request merged**